azure-connector/build.gradle.kts

```diff
@@ -0,0 +1,284 @@
+group = "io.aiven"
+
+// TODO: document why we stick to these versions
+val kafkaVersion by extra("1.1.0")
+val confluentPlatformVersion by extra("4.1.4")
+val avroConverterVersion by extra("7.2.2")
+val slf4jVersion by extra("1.7.36")
+val testcontainersVersion by extra("1.19.6")
+val wireMockVersion by extra("2.35.0")
+val mockitoVersion by extra("5.2.0")
+
+val integrationTest: SourceSet = sourceSets.create("integrationTest") {
+    java {
+        srcDir("src/integration-test/java")
+    }
+    resources {
+        srcDir("src/integration-test/resources")
+    }
+    compileClasspath += sourceSets["main"].output + configurations["testRuntimeClasspath"]
+    runtimeClasspath += output + compileClasspath
+}
+
+val integrationTestImplementation: Configuration by configurations.getting {
+    extendsFrom(configurations["implementation"])
+}
+
+tasks.register<Test>("integrationTest") {
+    description = "Runs the integration tests."
+    group = "verification"
+    testClassesDirs = integrationTest.output.classesDirs
+    classpath = integrationTest.runtimeClasspath
+
+    // defines testing order
+    shouldRunAfter("test")
+    // requires archive for connect runner
+    dependsOn("distTar")
+    useJUnitPlatform()
+
+    // Run always.
+    outputs.upToDateWhen { false }
+
+    // Pass the Azure connection string to the tests.
+    if (project.hasProperty("testAzureStorageString")) {
+        systemProperty("integration-test.azure.connection.string", project.findProperty("testAzureStorageString").toString())
+    }
+    // Pass the Azure container name to the tests.
+    systemProperty("integration-test.azure.container", project.findProperty("testAzureStorage").toString())
+    // Pass the distribution file path to the tests.
+    val distTarTask = tasks.getByName("distTar") as Tar
+    val distributionFilePath = distTarTask.archiveFile.get().asFile.path
+    systemProperty("integration-test.distribution.file.path", distributionFilePath)
+}
+
+idea {
+    module {
+        testSources.from(integrationTest.java.srcDirs)
+        testSources.from(integrationTest.resources.srcDirs)
+    }
+}
+
+dependencies {
+    compileOnly("org.apache.kafka:connect-api:$kafkaVersion")
+    compileOnly("org.apache.kafka:connect-runtime:$kafkaVersion")
+
+    implementation(project(":commons"))
+
+    implementation("com.azure:azure-storage-blob:12.26.1")
+    // TODO: document why specific version of guava is required
+    implementation("com.google.guava:guava:33.0.0-jre")
+
+    implementation("com.github.spotbugs:spotbugs-annotations:4.8.4")
+    implementation("org.slf4j:slf4j-api:$slf4jVersion")
+
+    testImplementation("org.junit.jupiter:junit-jupiter:5.10.2")
+    testImplementation("org.hamcrest:hamcrest:2.2")
+    testImplementation("org.assertj:assertj-core:3.25.3")
+    testImplementation("org.mockito:mockito-core:$mockitoVersion")
+    testImplementation("org.mockito:mockito-inline:$mockitoVersion")
+    testImplementation("net.jqwik:jqwik:1.8.4")
```

Comment:
JQWik creates a `azure-connector/.jqwik-database` file which should go to `.gitignore`.